### PR TITLE
Add API v2 using Django Rest Framework

### PR DESCRIPTION
**Pull request by secfigo**
This PR adds v2 of API using Django Rest Framework. 

1. The new code comes with 100% code coverage 
2. The new code is PEP8 compliant
3. Adds both Swagger UI and DRF browsable APIs.
4. Token Generation is also available via API.

To test it locally please follow the below steps

1. Setup Vagrant setup locally using the following Script from django-DefectDojo(obviously replace the existing Vagrantfile)

```bash
# -*- mode: ruby -*-
# vi: set ft=ruby :

# Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
    config.vm.box = "bento/ubuntu-16.04"
    config.vm.hostname = "defect-dojo"

    config.vm.network :private_network, ip: "192.168.13.37"

    # Exposes port 8000 on host from guest.
    config.vm.network "forwarded_port", guest: 8000, host: 8000

    # Shared folder for local code base
    config.vm.synced_folder ".", "/opt/defect-dojo/dojo/app/"

    config.vm.provider "virtualbox" do |v|
      v.memory = 1024
      v.name = "Defect-Dojo"
      v.cpus = 1
    end
end
```

2. login into vagrant box and run setup.bash script
```bash
vagrant ssh
cd /opt/defect-dojo/dojo/app/
./setup.bash  # select mysql, set root password and type it again when asked about MySQL details.
python manage.py runserver 0.0.0.0:8000
```
3.  Visit http://localhost:8000/ on your host machine, login and browse to v2 of api below
[http://localhost:8000/api/v2/](http://localhost:8000/api/v2/)

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is pep8 compliant (Dojo's code isn't currently pep8 compliant, but we're trying to correct that)
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation under the /docs folder

### CODE REVIEW

**Comment by aaronweaver** [comment]
@secfigo This looks great. I'll try this out and let you know if I have any issues.

File: dojo/models.py [hunk 1]
```diff
@@ -457,7 +461,7 @@ class IPScan(models.Model):
 
 
 class Engagement_Type(models.Model):
-    name = models.CharField(max_length=200)
+    name = models.CharField(max_length=200, primary_key=True)
```
Comment (aaronweaver) [inline dojo/models.py]: How would you resolve an issue if there is a duplicate engagement type?

**Comment by aaronweaver** [comment]
@secfigo Could you merge to the dev branch instead of master?

**Comment by maennel** [comment]
I think this is great work, @secfigo ! A big thumbs up.
One question: is there a specific reason the code in dojo/unittests/test_rest_framework.py is replicated in tests/rest_framework_test.py? From what I could see, the two files differ only in one line. 
If not needed, I would suggest to remove the file at tests/rest_framework_test.py in favour of the other, automatically executed one, in order to have the same code not appearing twice in the same repo.

File: dojo/models.py (hunk 1 above)
Comment (imohammedzen) [reply dojo/models.py]: I'm not aware if there will be such case. Can you please explain why/when that would happen ?

Comment (secfigo) [reply dojo/models.py]: I'm not aware, if there will be such a case. Can you please explain why/when that would happen ?

Comment (devGregA) [reply dojo/models.py]: Why change the primary key?

**Comment by devGregA** [comment]
@secfigo really like this, but I don't know how to auth for testing 😂 I think we could really use some docs.

File: dojo/models.py (hunk 1 above)
Comment (imohammedzen) [reply dojo/models.py]: @devGregA there is only one field in this model but will make it unique. DRF will throw an error if there is a duplicate engagement type but the bigger question is,  would there be any reason for engagement name to be duplicate  ?

**Comment by secfigo** [comment]
@devGregA Added instructions in the PR and also API v2 key under user profile.

**Comment by devGregA** [comment]
Awesome @secfigo! I'm good with merging. @aaronweaver good with you? @secfigo is there anything else you wanted to add before we merge? Or should I go ahead?

**Comment by devGregA** [comment]
Spoke to @aaronweaver merging.

**Pull request merged**